fastly/platform_tls.go

```diff
@@ -0,0 +1,350 @@
+	Name string `jsonapi:"attr,name,omitempty"`
+}
+
+// CreatePrivateKey create a TLS private key.
+func (c *Client) CreatePrivateKey(i *CreatePrivateKeyInput) (*PrivateKey, error) {
+
+	p := "/tls/private_keys"
+
+	if i.Key == "" {
+		return nil, ErrMissingKey
+	}
+
+	if i.Name == "" {
+		return nil, ErrMissingName
+	}
+
+	r, err := c.PostJSONAPI(p, i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var ppk PrivateKey
+	if err := jsonapi.UnmarshalPayload(r.Body, &ppk); err != nil {
+		return nil, err
+	}
+
+	return &ppk, nil
+}
+
+// DeletePrivateKeyInput used for deleting a private key.
+type DeletePrivateKeyInput struct {
+	ID string
+}
+
+// DeletePrivateKey destroy a TLS private key. Only private keys not already matched to any certificates can be deleted.
+func (c *Client) DeletePrivateKey(i *DeletePrivateKeyInput) error {
+	if i.ID == "" {
+		return ErrMissingID
+	}
+
+	path := fmt.Sprintf("/tls/private_keys/%s", i.ID)
+	_, err := c.Delete(path, nil)
+	return err
+}
+
+// BulkCertificate represents a bulk certificate.
+type BulkCertificate struct {
+	ID                string              `jsonapi:"primary,tls_bulk_certificate"`
+	TLSConfigurations []*TLSConfiguration `jsonapi:"relation,tls_configurations,tls_configuration"`
+	TLSDomains        []*TLSDomain        `jsonapi:"relation,tls_domains,tls_domain"`
+	NotAfter          *time.Time          `jsonapi:"attr,not_after,iso8601"`
+	NotBefore         *time.Time          `jsonapi:"attr,not_before,iso8601"`
+	CreatedAt         *time.Time          `jsonapi:"attr,created_at,iso8601"`
+	UpdatedAt         *time.Time          `jsonapi:"attr,updated_at,iso8601"`
+	Replace           bool                `jsonapi:"attr,replace"`
+}
+
+// TLSConfiguration represents the dedicated IP address pool that will be used to route traffic from the TLSDomain.
+type TLSConfiguration struct {
+	ID   string `jsonapi:"primary,tls_configuration"`
+	Type string `jsonapi:"attr,type"`
+}
+
+// TLSDomain represents a domain (including wildcard domains) that is listed on a certificate's Subject Alternative Names (SAN) list.
+type TLSDomain struct {
+	ID   string `jsonapi:"primary,tls_domain"`
+	Type string `jsonapi:"attr,type"`
+}
+
+// ListBulkCertificatesInput is used as input to the ListBulkCertificates function.
+type ListBulkCertificatesInput struct {
+	PageNumber *uint // The page index for pagination.
+	PageSize   *uint // The number of keys per page.
+	// `tls_domains.id` filter seems to work where `tls_domain.id` does not, documentation wrong?
+	// https://docs.fastly.com/api/platform-tls#tls_bulk_certificates_81cc5acbf847f71ecd4068ed58bfc5c5
+	FilterTLSDomainIDMatch  *string // Filter certificates by their matching, fully-qualified domain name. Returns all partial matches. Must provide a value longer than 3 characters.
+	FilterTLSDomainsIDMatch *string // Filter certificates by their matching, fully-qualified domain name. Returns all partial matches. Must provide a value longer than 3 characters.
```

Comment:
While both of these are supported, I'd prefer to have only the plural one included in this library as we work to have people migrate away from the singular filter name. Could I ask that you remove `FilterTLSDomainIDMatch` throughout and keep `FilterTLSDomainsIDMatch`?